eng/breakingChanges/breaking-change-doc.ps1

```diff
@@ -0,0 +1,1114 @@
+}
+
+# Collect detailed data for each PR
+$analysisData = @()
+foreach ($pr in $prs) {
+    Write-Host "   Collecting data for PR #$($pr.number): $($pr.title)" -ForegroundColor Gray
+
+    # Get comprehensive PR details including comments and reviews
+    try {
+        $prDetails = gh pr view $pr.number --repo $Config.SourceRepo --json body,title,comments,reviews,closingIssuesReferences
+        $prDetailData = $prDetails | ConvertFrom-Json
+    } catch {
+        Write-Warning "Could not fetch detailed PR data for #$($pr.number)"
+        continue
+    }
+
+    # Get closing issues with full details and comments
+    $closingIssues = @()
+    foreach ($issueRef in $prDetailData.closingIssuesReferences) {
+        if ($issueRef.number) {
+            try {
+                Write-Host "     Fetching issue #$($issueRef.number)..." -ForegroundColor DarkGray
+                $issueDetails = gh issue view $issueRef.number --repo $Config.SourceRepo --json number,title,body,comments,labels,state,createdAt,closedAt,url
+                $issueData = $issueDetails | ConvertFrom-Json
+                $closingIssues += @{
+                    Number = $issueData.number
+                    Title = $issueData.title
+                    Body = $issueData.body
+                    Comments = $issueData.comments
+                    Labels = $issueData.labels | ForEach-Object { $_.name }
+                    State = $issueData.state
+                    CreatedAt = $issueData.createdAt
+                    ClosedAt = $issueData.closedAt
+                    Url = $issueData.url
+                }
+            }
+            catch {
+                Write-Warning "Could not fetch issue #$($issueRef.number)"
+            }
+        }
+    }
+
+    # Create merge commit URL
+    $mergeCommitUrl = if ($pr.mergeCommit.oid) {
+        "https://github.com/$($Config.SourceRepo)/commit/$($pr.mergeCommit.oid)"
+    } else {
+        $null
+    }
+
+    # Get version information using local git repository
+    Write-Host "     🏷️ Getting version info..." -ForegroundColor DarkGray
+    $versionInfo = Get-VersionInfo -prNumber $pr.number -mergedAt $pr.closedAt -baseRef $pr.baseRefName
+
+    # Check for existing docs issues
+    $hasDocsIssue = $false
+    try {
+        $searchResult = gh issue list --repo $Config.DocsRepo --search "Breaking change $($pr.number)" --json number,title
+        $existingIssues = $searchResult | ConvertFrom-Json
+        $hasDocsIssue = $existingIssues.Count -gt 0
+    } catch {
+        Write-Warning "Could not check for existing docs issues for PR #$($pr.number)"
+    }
+
+    # Get feature areas from area- labels first, then fall back to file paths
+    $featureAreas = @()
+
+    # First try to get feature areas from area- labels
+    foreach ($label in $pr.labels) {
+        if ($label.name -match "^area-(.+)$") {
+            $featureAreas += $matches[1]
+        }
+    }
+
+    # If no area labels found, fall back to file path analysis
```

Comment:
What's the scenario for wanting to determine feature areas without area labels? I've found it's pretty time-consuming for the GH API to get the list of files from PRs, especially for PRs older that then newest 100 (presumably due to caching optimizations), so if we can avoid collecting the file names that would be beneficial.

Reply:
Got it; thanks. I found the `dotnet-api-docs` repo as a good stress test for crawling PRs, and other repos where hundreds of files are edited at once on PRs commonly also exhibit the perf cliff I've seen. In order for [labeler-train.yml (dotnet/dotnet-api-docs)](https://github.com/dotnet/dotnet-api-docs/blob/main/.github/workflows/labeler-train.yml#L37) to work, it can only consume 1 pull request at a time and even with that, GitHub's API sometimes times out.